File: OTCDerivativesCalculatorModule/Project_CSharp/XmlBuilder/XmlSerializedClass/fpml-shared-5-4/Stub.cs
using System;
using System.Text;
using System.Collections.Generic;
using System.Xml;

namespace FpmlSerializedCSharp
{
    public class Stub : StubValue
    {
        public Stub(XmlNode xmlNode)
        : base(xmlNode)
        {
            XmlNode stubStartDateNode = xmlNode.SelectSingleNode("stubStartDate");
            
            if (stubStartDateNode != null)
            {
                if (stubStartDateNode.Attributes["href"] != null || stubStartDateNode.Attributes["id"] != null) 
                {
                    if (stubStartDateNode.Attributes["id"] != null) 
                    {
                        stubStartDateIDRef_ = stubStartDateNode.Attributes["id"].Value;
                        AdjustableOrRelativeDate ob = new AdjustableOrRelativeDate(stubStartDateNode);
                        IDManager.SetID(stubStartDateIDRef_, ob);
                    }
                    else if (stubStartDateNode.Attributes["href"] != null)
                    {
                        stubStartDateIDRef_ = stubStartDateNode.Attributes["href"].Value;
                    }
                    else
                    {
                        stubStartDate_ = new AdjustableOrRelativeDate(stubStartDateNode);
                    }
                }
                else
                {
                    stubStartDate_ = new AdjustableOrRelativeDate(stubStartDateNode);
                }
            }
            
        
            XmlNode stubEndDateNode = xmlNode.SelectSingleNode("stubEndDate");
            
            if (stubEndDateNode != null)
            {
                if (stubEndDateNode.Attributes["href"] != null || stubEndDateNode.Attributes["id"] != null) 
                {
                    if (stubEndDateNode.Attributes["id"] != null) 
                    {
                        stubEndDateIDRef_ = stubEndDateNode.Attributes["id"].Value;
                        AdjustableOrRelativeDate ob = new AdjustableOrRelativeDate(stubEndDateNode);
                        IDManager.SetID(stubEndDateIDRef_, ob);
                    }
                    else if (stubEndDateNode.Attributes["href"] != null)
                    {
                        stubEndDateIDRef_ = stubEndDateNode.Attributes["href"].Value;
                    }
                    else
                    {
                        stubEndDate_ = new AdjustableOrRelativeDate(stubEndDateNode);
                    }
                }
                else
                {
                    stubEndDate_ = new AdjustableOrRelativeDate(stubEndDateNode);
                }
            }
            
        
        }
        
    
        #region StubStartDate_
        private AdjustableOrRelativeDate stubStartDate_;
        public AdjustableOrRelativeDate StubStartDate_
        {
            get
            {
                if (this.stubStartDate_ != null)
                {
                    return this.stubStartDate_; 
                }
                else if (this.stubStartDateIDRef_ != null)
                {
                    stubStartDate_ = IDManager.getID(stubStartDateIDRef_) as AdjustableOrRelativeDate;
                    return this.stubStartDate_; 
                }
                else
                {
                      return this.stubStartDate_; 
                }
            }
            set
            {
                if (this.stubStartDate_ != value)
                {
                    this.stubStartDate_ = value;
                }
            }
        }
        #endregion
        
        public string stubStartDateIDRef_ { get; set; }
        #region StubEndDate_
        private AdjustableOrRelativeDate stubEndDate_;
        public AdjustableOrRelativeDate StubEndDate_
        {
            get
            {
                if (this.stubEndDate_ != null)
                {
                    return this.stubEndDate_; 
                }
                else if (this.stubEndDateIDRef_ != null)
                {
                    stubEndDate_ = IDManager.getID(stubEndDateIDRef_) as AdjustableOrRelativeDate;
                    return this.stubEndDate_; 
                }
                else
                {
                      return this.stubEndDate_; 
                }
            }
            set
            {
                if (this.stubEndDate_ != value)
                {
                    this.stubEndDate_ = value;
                }
            }
        }
        #endregion
        
        public string stubEndDateIDRef_ { get; set; }
        
    
        
    
    }
    
}

File: OTCDerivativesCalculatorModule/Project_Cpp/lib_static/FpmlSerialized/GenClass/fpml-shared-5-4/Stub.cpp
// Stub.cpp 
#include "Stub.hpp"
#ifdef ConsolePrint
    #include <iostream>
#endif 

namespace FpmlSerialized {

Stub::Stub(TiXmlNode* xmlNode)
: StubValue(xmlNode)
{
    #ifdef ConsolePrint
        std::string initialtap_ = FileManager::instance().tap_;
        FileManager::instance().tap_.append("   ");
    #endif 
   //stubStartDateNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* stubStartDateNode = xmlNode->FirstChildElement("stubStartDate");

   if(stubStartDateNode){stubStartDateIsNull_ = false;}
   else{stubStartDateIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- stubStartDateNode , address : " << stubStartDateNode << std::endl;
   #endif
   if(stubStartDateNode)
   {
      if(stubStartDateNode->Attribute("href") || stubStartDateNode->Attribute("id"))
      {
          if(stubStartDateNode->Attribute("id"))
          {
             stubStartDateIDRef_ = stubStartDateNode->Attribute("id");
             stubStartDate_ = boost::shared_ptr<AdjustableOrRelativeDate>(new AdjustableOrRelativeDate(stubStartDateNode));
             stubStartDate_->setID(stubStartDateIDRef_);
             IDManager::instance().setID(stubStartDateIDRef_,stubStartDate_);
          }
          else if(stubStartDateNode->Attribute("href")) { stubStartDateIDRef_ = stubStartDateNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { stubStartDate_ = boost::shared_ptr<AdjustableOrRelativeDate>(new AdjustableOrRelativeDate(stubStartDateNode));}
   }

   //stubEndDateNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* stubEndDateNode = xmlNode->FirstChildElement("stubEndDate");

   if(stubEndDateNode){stubEndDateIsNull_ = false;}
   else{stubEndDateIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- stubEndDateNode , address : " << stubEndDateNode << std::endl;
   #endif
   if(stubEndDateNode)
   {
      if(stubEndDateNode->Attribute("href") || stubEndDateNode->Attribute("id"))
      {
          if(stubEndDateNode->Attribute("id"))
          {
             stubEndDateIDRef_ = stubEndDateNode->Attribute("id");
             stubEndDate_ = boost::shared_ptr<AdjustableOrRelativeDate>(new AdjustableOrRelativeDate(stubEndDateNode));
             stubEndDate_->setID(stubEndDateIDRef_);
             IDManager::instance().setID(stubEndDateIDRef_,stubEndDate_);
          }
          else if(stubEndDateNode->Attribute("href")) { stubEndDateIDRef_ = stubEndDateNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { stubEndDate_ = boost::shared_ptr<AdjustableOrRelativeDate>(new AdjustableOrRelativeDate(stubEndDateNode));}
   }

    #ifdef ConsolePrint
        FileManager::instance().tap_ = initialtap_;
    #endif 
}
boost::shared_ptr<AdjustableOrRelativeDate> Stub::getStubStartDate()
{
   if(!this->stubStartDateIsNull_){
        if(stubStartDateIDRef_ != ""){
             return boost::shared_static_cast<AdjustableOrRelativeDate>(IDManager::instance().getID(stubStartDateIDRef_));
        }else{
             return this->stubStartDate_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<AdjustableOrRelativeDate>();
   }
}
boost::shared_ptr<AdjustableOrRelativeDate> Stub::getStubEndDate()
{
   if(!this->stubEndDateIsNull_){
        if(stubEndDateIDRef_ != ""){
             return boost::shared_static_cast<AdjustableOrRelativeDate>(IDManager::instance().getID(stubEndDateIDRef_));
        }else{
             return this->stubEndDate_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<AdjustableOrRelativeDate>();
   }
}
}

